server/api/apiSpecification.yaml

```diff
@@ -0,0 +1,568 @@
+    # // GET /getAllAttributes/604f537ed699a2eb47433184'
+    get:
+      tags:
+        - Robots
+      summary: Get all rpa attribute objects for a specific robot
+      operationId: getAttributesForRobot
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/RPAAttributes'
+components:
+  schemas:
+    #### General ####
+    ObjectIds:
+      type: string
+      minLength: 1
+      example: 6061d5627dee5516b3824ade
+    #### RPA-Functionalities ####
+    Functionalities:
+      type: object
+      required:
+        - _id
+        - Application
+        - Task
+        - Code
+      properties:
+        _id:
+          $ref: '#/components/schemas/ObjectIds'
+        Application:
+          $ref: '#/components/schemas/Applications'
+        Task:
+          $ref: '#/components/schemas/Tasks'
+        Code:
+          $ref: '#/components/schemas/Tasks'
+        ouputValue:
+          type: boolean
+        inputVars:
+          type: object
+          required:
+            - name
+            - type
+            - isRequired
+            - infoText
+            - index
+          properties:
+            name:
+              type: string
+              example: visible
+            type:
+              type: string
+              example: Boolean
+            isRequired:
+              type: boolean
+            infoText:
+              type: string
+              example: Show window after opening
+            index:
+              type: integer
+              example: 0
+    Applications:
+      type: string
+      example: Excel.Files
+    Tasks:
+      type: string
+      example: Open Workbook
+    #### Robots ####
+    RobotCode:
+      type: string
+    RPAAttributes:
+      type: object
+      required:
+        - robotId
+        - activityId
+        - rpaApplication
+        - rpaTask
+      properties:
+        robotId:
+          $ref: '#/components/schemas/RobotIds'
+        activityId:
+          $ref: '#/components/schemas/ActivityIds'
+        rpaApplication:
+          $ref: '#/components/schemas/Applications'
+        rpaTask:
+          $ref: '#/components/schemas/Tasks'
+    Parameters:
+      type: object
+      required:
+        - activityId
+        - robotId
+      properties:
+        activityId:
+          $ref: '#/components/schemas/ActivityIds'
+        robotId:
+          $ref: '#/components/schemas/RobotIds'
+        outputVariable:
+          type: string
+          example: Activity_0n140xq_output
+        rpaParameters:
+          type: array
+          items:
+            $ref: '#/components/schemas/singleParameterObjects'
+    singleParameterObjects:
+      type: object
+      required:
+        - name
+        - value
+        - requireUserInput
+        - type
+        - isRequired
+        - infoText
+        - index
```

Comment:
This means as soon as we change a model, we need to change this too?

Reply:
yes